### ISSUE TITLE
[a11y]: DataTable [batch actions] - VO extraneous announcement 

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14259]
### Package

@carbon/react

### Browser

Safari & Iphone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO 

### Description
Select all Rows check box in the first row of the table

VO extraneous announcement after checking/unchecking the check box in the column header

VO announces the name of this check box multiple times after when we select / unselect.

### Recommended Fix
remove the extraneous announcement

Any redundant, excessively wordy and extraneous announcement can be distracting for the screen reader users. 

### WCAG 2.1 Violation

4.1.2 Name, Role, Value

### Reproduction/example
Record `Number 62` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems](https://github.com/carbon-design-system/carbon/issues/14255)

**Comment by francinelucca** [comment]
unchecking/checking “Select all” makes VoiceOver announce multiple times “select row, select row, select row, select row”

**Comment by guidari** [comment]
I tested this issue in the Safari and now it is working. It only says one time "select row".
